### PR TITLE
[pub/sub] Clarify at-least-once and consumer groups

### LINKED ISSUE #3435: Document that some PubSub components do not implement Competing Consumer model
**Describe the issue**

[Originating from this conversation in Discord](https://discord.com/channels/778680217417809931/901141467840524289/1108140265065168897)

PubSub component implementations have become more diverse over time. This means that not all PubSub components have a competing consumer model, however the docs imply that all components have a competing consumer mechanism.

**URL of the docs**

https://docs.dapr.io/developing-applications/building-blocks/pubsub/pubsub-overview/#at-least-once-guarantee

**Expected content**

In order to remove anxiety, the docs should make confident remarks to the at-least-once guarantee for all components and demonstrate **how** the at-least-once guarantee differs for competing consumers Vs parallel receivers.

The docs should state **which** components use true Competing Consumers Vs components which deliver the message in parallel to all replicas in that consumer group / app-id. 

**Screenshots**
N/A

**Additional context**

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

Try to clarify at-least-once guarantee and the consumer groups sections. 

@msfussell is there a way to find out which pub/sub components support the competing consumers pattern?

## Issue reference

PR will close: #3435

### CODE REVIEW

**Comment by msfussell** [comment]
We should clarify which pub/sub components do not support the competing consumer pattern.

**Comment by hhunter-ms** [comment]
> We should clarify which pub/sub components do not support the competing consumer pattern.

@msfussell do we know this list? in the Discord discussion linked in the related issue, the only ones we know support it are Azure Service Bus Queues, Redis, RabbitMQ, EventGrid, and Apache Kafka - and even that hasn't been fully confirmed

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-overview.md [hunk 1]
```diff
@@ -100,16 +100,29 @@ Dapr solves multi-tenancy at-scale with [namespaces for consumer groups]({{< ref
...
+
+When multiple instances of the same application (with same `app-id`) subscribe to a topic, Dapr delivers each message to *only one instance of **that** application*. This concept is illustrated in the diagram below.
 
 <img src="/images/pubsub-overview-pattern-competing-consumers.png" width=1000>
 <br></br>
 
-Similarly, if two different applications (with different app-IDs) subscribe to the same topic, Dapr delivers each message to *only one instance of **each** application*.
+Similarly, if two different applications (with different `app-id`) subscribe to the same topic, Dapr delivers each message to *only one instance of **each** application*.
+
+Not all Dapr pub/sub components support the competing consumer model. Currently, the following pub/sub components support this pattern:
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-overview.md]:
```suggestion
Not all Dapr pub/sub components support the competing consumer pattern. Currently, the following (non-exhaustive) pub/sub components support this:
```

**Review by msfussell — APPROVED** [review]
LTGM with my comment suggestion

**Pull request merged**